src/prisma/schema.prisma

```diff
@@ -97,14 +110,34 @@ model WalletDetails {
   gcpKmsResourcePath                 String? @map("gcpKmsResourcePath") @db.Text
   gcpApplicationCredentialEmail      String? @map("gcpApplicationCredentialEmail") /// if not available, default to: Configuration.gcpApplicationCredentialEmail
   gcpApplicationCredentialPrivateKey String? @map("gcpApplicationCredentialPrivateKey") /// if not available, default to: Configuration.gcpApplicationCredentialPrivateKey
+
   // Smart Backend Wallet
-  accountSignerAddress               String? @map("accountSignerAddress") /// this, and either local, aws or gcp encryptedJson, are required for smart wallet
-  accountFactoryAddress              String? @map("accountFactoryAddress") /// optional even for smart wallet, if not available default factory will be used
-  entrypointAddress                  String? @map("entrypointAddress") /// optional even for smart wallet, if not available SDK will use default entrypoint
+  accountSignerAddress  String? @map("accountSignerAddress") /// this, and either local, aws or gcp encryptedJson, are required for smart wallet
+  accountFactoryAddress String? @map("accountFactoryAddress") /// optional even for smart wallet, if not available default factory will be used
+  entrypointAddress     String? @map("entrypointAddress") /// optional even for smart wallet, if not available SDK will use default entrypoint
 
   @@map("wallet_details")
 }
 
+model WalletCredentials {
+  id        String  @id @default(uuid()) @map("id")
+  type      String  @map("type")
+  label     String  @map("label")
+  data      Json    @map("data")
+  isDefault Boolean @default(false) @map("isDefault")
```

Comment:
IMO Engine doesn't need to support more than one credential per type. It seems like the added bloat of handling the edge cases around "default vs non-default" credentials is not worth being able to support wallets from multiple AWS accounts or Circle accounts on the same Engine.

Reply:
I think it's worth having, especially if we want to make this the core for engine-cloud. Also, the WalletCredentials system really only shines if we have multiple credentials. A `json` field on Config would have sufficed otherwise 😅